internal/core/metadata/v2/controller/http/deviceprofile.go

```diff
@@ -321,3 +321,47 @@ func (dc *DeviceProfileController) DeleteDeviceProfileByName(w http.ResponseWrit
 	utils.WriteHttpHeader(w, ctx, statusCode)
 	pkg.Encode(response, w, lc)
 }
+
+func (dc *DeviceProfileController) GetAllDeviceProfiles(w http.ResponseWriter, r *http.Request) {
+	lc := container.LoggingClientFrom(dc.dic.Get)
+	ctx := r.Context()
+	correlationId := correlation.FromContext(ctx)
+
+	var response interface{}
+	var statusCode int
+
+	// parse URL query string
+	offset, err := utils.ParseQueryStringToInt(r, contractsV2.Offset, contractsV2.DefaultOffset)
```

Comment:
Does this need the same changes made in PR #2805 ?

Reply:
Yes. 
Made changes according to the PR.